src/components/appheader.main.jsx

```diff
@@ -124,35 +127,38 @@ class AppHeaderMain extends React.Component {
 
           <div className="central-container">
             <div className="horizontal-menu">
+              {availability && (
               <AppHeaderNavigationMain isOfflineCheck={this.handleIsOffline} isOffline={isOffline} isMobileView={false} />
+              )}
             </div>
           </div>
 
-          {(!Config.b2b.enable || (Config.b2b.enable && isB2bCartSelected)) && (
+          {(!Config.b2b.enable || (Config.b2b.enable && isB2bCartSelected && availability)) && (
             <div className="search-container">
               <AppHeaderSearchMain isMobileView={false} />
             </div>
           )}
-
-          <div className="search-toggle-btn-container">
-            <button
-              className="search-toggle-btn"
-              type="button"
-              data-toggle="collapse"
-              data-target=".collapsable-container"
-              aria-expanded="false"
-              aria-label="Toggle navigation"
-              onClick={this.handleInputFocus}
-            >
-              <div className="search-icon" />
-            </button>
-          </div>
+          {availability && (
```

Comment:
We shouldn't hide the search altogether here. The `CATALOG_BROWSER` role has access to browse the catalog, but cannot add to cart or wishlists. So on the Product Display Page, we should disable those buttons if the forms are not returned by the API.